src/components/TextInput/TextInput.jsx

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { v4 as uuid } from 'uuid';
+import Cleave from 'cleave.js/react';
+import * as InputMasks from './TextInputMasks';
+import './TextInput.scss';
+import FormLabel from '../FormLabel/FormLabel';
+
+const getInputMask = (inputMask, availableInputMasks) => {
+  if (typeof inputMask === 'string') {
+    return availableInputMasks[inputMask];
+  }
+
+  return inputMask;
+};
+
```

Comment:
```suggestion
/**
 * Use TextInput to show where users can enter text based data. It does not maintain any internal state, so its value should be managed by the parent.
 */
```

Add description for documentation. Feel free to alter if this is right.